### ISSUE TITLE
Updates and replaces fail silently on bulk write

### ISSUE THREAD

**Issue by laplaceon** [mongodb-labs/mongo-rust-driver-prototype#257]
I have tried to use the bulk write method for collections, but writemodels of variant replaceone and updateone are silently failing while insertone works. I checked my mongod log files and there is no evidence of an error. I am using rustc 1.26.2 and mongodb 3.6.4.

**Comment by laplaceon** [comment]
I'm not completely familiar with the crate, but after some inspection of the source code, I found snippets of code that seem suspect and may relate to this issue.

On lines 606-608 of coll/mod.rs: https://github.com/mongodb-labs/mongo-rust-driver-prototype/blob/5b9807e8b1e54cb89d318f42de8c45264da55e5e/src/coll/mod.rs#L606-L608

Shouldn't this be:
`        if !exception.unprocessed_requests.is_empty() {`
If the unprocessed_requests vector is empty, wouldn't that mean no unprocessed requests?

On lines 539-543 of coll/mod.rs: https://github.com/mongodb-labs/mongo-rust-driver-prototype/blob/5b9807e8b1e54cb89d318f42de8c45264da55e5e/src/coll/mod.rs#L539-L543

These lines seem to be copied from execute_delete_batch and it's strange to me that they are in a function called execute_update_batch. Is this a typo or is there an ulterior motive?

I cloned the repo and changed the lines above and re-ran my app, but the update operations still do not go through. However, the BulkWriteException now is correctly returned as Some and my updates are present in the unprocessed requests vector.
